File: 0053-maximum-subarray/0053-maximum-subarray.py
import sys
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0]
        s=0
        for i in nums:
            s+=i
            if(s>maxSum):
                maxSum = s
            if(s<0):
                s=0
        return maxSum

        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=0;
        int maxSum = nums[0];
        for(int i:nums){
            sum+=i;
            if(maxSum<sum) maxSum = sum;
            if(sum<0) sum=0;
        }
        return maxSum;
    }
};